### PR TITLE
feat: replace old portfolio

### PR DESCRIPTION
**Pull request by 120EE0692**
### Your checklist for this pull request

🚨Please review the [guidelines for contributing](https://github.com/Monday-Morning/project-tahiti/blob/main/CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your default branch!
- [x] Make sure you are making a pull request against the **default branch** (left side). Also you should start _your branch_ off _default branch_.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.
- [x] I have added necessary documentation (if appropriate)

### Description
Activate the user old portfolio page in new website

❤️Thank you!

### Post merge checklist

- [ ] Follow steps from the [guidelines for contributing](https://github.com/Monday-Morning/project-tahiti/blob/main/CONTRIBUTING.md) to this repository.
- [ ] If you are a new contributor, ping in the thread and one of the maintainers will add you to all-contributors list.

### CODE REVIEW

File: client/src/pages/portfolio/[userId].jsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    },
+  } = user;
+
+  const noProfilePage = ['Guest', 'Team'];
+
+  if (noProfilePage.includes(firstName)) {
+    return {
+      notFound: true,
+    };
+  }
```
Comment (rutajdash) [inline client/src/pages/portfolio/[userId].jsx:84]: Let a profile exist for both these accounts

**Review by rutajdash — CHANGES_REQUESTED** [review]
Ensure the following:
1. If someone opens a profile page which is `/portfolio/<newUserId>`, redirect it to `/portfolio/<newUserId>/<generatedUserName`
2. If someone opens a profile page which is `/portfolio/<newUserId>/<invalidUserName>`, redirect it to `/portfolio/<newUserId>/<generatedUserName`
3. The generatedUserName must be `<firstName>_<lastName>`
4. If anyone opens `/portfolio/<oldUserName>`, redirect it to `/portfolio/<newUserId>/<generatedUserName`
5. If anyone has an Email Id as `transfer-<something>@gmail.com`, do not show any email ID.

File: client/src/pages/portfolio/[userId].jsx (hunk 1 above)
Comment (120EE0692) [reply client/src/pages/portfolio/[userId].jsx:84]: Such profiles have 150+ articles. We have to implement pagination for this. Which I planned to do later.

File: client/src/pages/portfolio/[userId]/[name].jsx [hunk 2]
```diff
@@ -93,6 +91,8 @@ export async function getStaticProps({ params: { userId } }) {
     },
   });
 
+  const noOfArticle = articleList.length;
```
[the remark this answers is not in the document]
Comment (120EE0692) [reply client/src/pages/portfolio/[userId]/[name].jsx:94]: I made a wrong assumption that a author have not written more than 40 articles. 
And hence few of the portfolio are not working.

**Comment by rutajdash** [comment]
@120EE0692 Tag me and let me know once the following is done.

> Ensure the following:
> 
> 1. If someone opens a profile page which is `/portfolio/<newUserId>`, redirect it to `/portfolio/<newUserId>/<generatedUserName`
> 2. If someone opens a profile page which is `/portfolio/<newUserId>/<invalidUserName>`, redirect it to `/portfolio/<newUserId>/<generatedUserName`
> 3. The generatedUserName must be `<firstName>_<lastName>`
> 4. If anyone opens `/portfolio/<oldUserName>`, redirect it to `/portfolio/<newUserId>/<generatedUserName`
> 5. If anyone has an Email Id as `transfer-<something>@gmail.com`, do not show any email ID.

**Comment by rutajdash** [comment]
@120EE0692 please complete the mentioned tasks.

**Review by rutajdash — CHANGES_REQUESTED** [review]
1. User slug check logic is inefficient.
2. Unnecessary details in getUserByOldUserName

**Review by rutajdash — APPROVED** [review]

**Pull request merged**